### ISSUE TITLE
YOLO v4 for VPU

### ISSUE THREAD

**Issue by ankandrew** [AlexeyAB/darknet#5467]
In the paper I noticed that for VPU some good candidates are EfficientNet-lite, MixNet, GhostNet and MobileNetV3. Did it gave good results these as backbones for YOLO v4 and, decrease the inference time considerably?

**Comment by AlexeyAB** [comment]
EfficientNetB0-Yolo shows good result AP (MSCOCO) and 11 FPS on VPU even for: 416x416, batch=1, async=1: https://github.com/AlexeyAB/darknet/issues/5079

So we should try to use EfficientNet-b3-lite without SE, and and should use neck/head without PRN (because it slows down inference on VPU).

----

MixNet shows better accuracy/BFLOPS than EfficientNetB0, but another question is how well does it can be parallelized: https://github.com/AlexeyAB/darknet/issues/4203

May be GhostNet is more optimal than MobieNetv3.

**Comment by ankandrew** [comment]
It would also be interesting maybe to consider a pruned version of EfficientNet (https://arxiv.org/abs/2002.08258) Apparently "A pruned EfficientNet-B1 may be more efficient than EfficientNet-B0." Maybe this can push AP a little further.
Also, I guess the EfficientNet-b3-lite-YOLO option for VPUs might be interesting (with ReLU6 maybe). There is also [EfficientNets-Edge TPU](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet/edgetpu) optimized specifically for the Coral Edge TPU, which might be very good in this specific case (don't know how it may perform on other VPU).

Thank you for the response

**Comment by AlexeyAB** [comment]
Pruning usually gives the same results as training with initially fewer weights. And in the EfficientNet, the optimal number of weights has already been chosen using  AutoML MNAS.

ReLU6 - should be used only if you want to quantize it to INT8.

**Comment by ankandrew** [comment]
Yes, looking to the comparison of EfficientNet B1 Knapsack Pruned and B0, looks like it is doing only a little bit better than the MNAS own model. Guess this won't contribute anything

![Annotation 2020-05-04 131456](https://user-images.githubusercontent.com/61120139/80988095-5f1ff080-8e09-11ea-9799-f5f25f7938d0.png)

As my original question was answered, I am closing it
Thanks

**Comment by AlexeyAB** [comment]
@ankandrew It seems there was something wrong during first testing of EfficientNetB0-Yolo on VPU. I re-tested many models. So it seems that EfficientNet isn't suitable even for VPU )


* For EfficientNet-Yolo (Leaky instead of Swith, and without SE) `256x256 async=3` – I get oly 12.5 FPS 
* Approximately the same speed I get by using YOLOv4 (Leaky) `256x256 async=3` - 11 FPS, but accuracy much higher than in the EfficientNet-Yolo.

So even YOLOv4 (Leaky) more suitable for VPU than EfficientNet.

**Comment by ankandrew** [comment]
I wouldn't have exptected that bad performance of EfficientNet-Yolo, any thoughts why YOLO v4 is performing much better in accuracy?

**Comment by AlexeyAB** [comment]
@ankandrew Are you abount accuracy or FPS?

About accuracy:
* YOLOV4 (with Leaky, 256x256) gives `53.0%` AP50 (checked on test-dev) - 11 FPS (async=3)

* EfficientNetB0-Yolo (with SE, Swish, 416x416) gives only `45.5%` AP50 - just imagine how much accuracy will be for EfficientNetB0-Yolo (**without SE, without Swish, 256x256**) - something like `~40%` AP50 - 12 FPS (async=3)

* EfficientNetB0+BiFPN-official (with SE, Swish, 416x416) gives only `52.2%` AP50 - just imagine how much accuracy will be for EfficientNetB0+BiFPN-official (**without SE, without Swish, 256x256**) - something like `~45%` AP50

That's why Google doesnt use Swish, SE, Grouped-conv for TPU-edge: https://ai.googleblog.com/2019/08/efficientnet-edgetpu-creating.html

**Comment by ankandrew** [comment]
Interesting, YOLOV4 (with Leaky, 256x256) outstands EfficientNetB0-Yolo in accuracy and EfficientNetB0-Yolo (with SE, Swish, 416x416) and guess it is much faster than the latter one.
Just for curious are there any plans to implement YOLO v4 specially for Coral Edge TPU?
I am curious on how a fully int8 quantized model on the Coral TPU can be compared to a model in Intel Myriad X (FP16) in terms of speed & accuracy.

Thanks for the update 👍

**Comment by vinorth05** [comment]
> Just for curious are there any plans to implement YOLO v4 specially for Coral Edge TPU?
> 
> Thanks for the update

I am also interested!
